From b4e4d9a96619f0c818c6cf7c6aeef8c564494dd1 Mon Sep 17 00:00:00 2001
From: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Date: Thu, 14 Jul 2022 10:14:39 +0200
Subject: [PATCH] dt-bindings: Add various vendor-prefixes and devices

- Document the bytesatwork AG vendor prefix.
- Document the youritech vendor prefix.
- Add bindings for byteENGINEs and byteDEVKITs.
- Add the bindings for the MIPI-DSI display from Youritech with the
  ILI9806E controller.

Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Signed-off-by: Daniel Ammann <daniel.ammann@bytesatwork.ch>
---
 .../bindings/arm/bytesatwork/bytesatwork.yaml | 37 ++++++++
 .../display/panel/youritech,ili9806e.yaml     | 84 +++++++++++++++++++
 .../devicetree/bindings/vendor-prefixes.yaml  |  4 +
 MAINTAINERS                                   |  2 +
 4 files changed, 127 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
 create mode 100644 Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml

diff --git a/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml b/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
new file mode 100644
index 000000000000..04281eb5dcf4
--- /dev/null
+++ b/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/arm/bytesatwork/bytesatwork.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: bytesatwork Platforms Device Tree Bindings
+
+maintainers:
+  - Daniel Ammann <daniel.ammann@bytesatwork.ch>
+
+properties:
+  $nodename:
+    const: "/"
+  compatible:
+    oneOf:
+      - description: bytesatwork byteENGINE
+        items:
+          - enum:
+              - bytesatwork,byteengine-am62x
+              - bytesatwork,byteengine-imx8mm
+              - bytesatwork,byteengine-imx8mp
+              - bytesatwork,byteengine-imx93
+              - bytesatwork,byteengine-stm32mp1
+
+      - description: bytesatwork byteDEVKIT
+        items:
+          - enum:
+              - bytesatwork,bytedevkit-am62x
+              - bytesatwork,bytedevkit-imx8mm
+              - bytesatwork,bytedevkit-imx8mp
+              - bytesatwork,bytedevkit-imx93
+              - bytesatwork,bytedevkit-stm32mp1
+
+additionalProperties: true
+
+...
diff --git a/Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml b/Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml
new file mode 100644
index 000000000000..9c5eb2bf2e1f
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/youritech,ili9806e.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: MIPI-DSI panel with Youritech ILI9806E Controller
+
+maintainers:
+  - Daniel Ammann <daniel.ammann@bytesatwork.ch>
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    const: youritech,ili9806e
+
+  reg: true
+  port: true
+  reset-gpios: true
+  width-mm: true
+  height-mm: true
+
+  dsi-lanes:
+    description: Number of DSI lanes to be used must be <1> or <2> or <3> or <4>
+    enum: [1, 2, 3, 4]
+
+  video-mode:
+    description: |
+      0 - burst-mode
+      1 - non-burst with sync event
+      2 - non-burst with sync pulse
+      3 - command mode
+    $ref: /schemas/types.yaml#/definitions/uint32
+    enum: [0, 1, 2, 3]
+
+required:
+  - compatible
+  - reg
+  - dsi-lanes
+  - port
+
+additionalProperties: false
+
+examples:
+  - |
+    &dsi {
+      #address-cells = <1>;
+      #size-cells = <0>;
+      status = "okay";
+
+      panel@0 {
+        compatible = "youritech,ili9806e";
+        reg = <0>;
+        backlight = <&mipi_bl>;
+        reset-gpios = <&gpio2 22 GPIO_ACTIVE_HIGH>;
+        dsi-lanes = <2>;
+        video-mode = <2>;	/* 0: burst mode
+              * 1: non-burst mode with sync event
+              * 2: non-burst mode with sync pulse
+              */
+        width-mm = <68>;
+        height-mm = <121>;
+        status = "okay";
+
+        port {
+          panel_mipi_in: endpoint {
+            remote-endpoint = <&dsi_out>;
+          };
+        };
+      };
+
+      ports {
+        port@1 {
+          reg = <1>;
+
+          dsi_out: endpoint {
+            remote-endpoint = <&panel_mipi_in>;
+          };
+        };
+      };
+    };
+...
diff --git a/Documentation/devicetree/bindings/vendor-prefixes.yaml b/Documentation/devicetree/bindings/vendor-prefixes.yaml
index 336598c3b6f8..f6e1ac0d0bb7 100644
--- a/Documentation/devicetree/bindings/vendor-prefixes.yaml
+++ b/Documentation/devicetree/bindings/vendor-prefixes.yaml
@@ -210,6 +210,8 @@ patternProperties:
     description: B&R Industrial Automation GmbH
   "^bytedance,.*":
     description: ByteDance Ltd.
+  "^bytesatwork,.*":
+    description: bytesatwork AG
   "^calamp,.*":
     description: CalAmp Corp.
   "^calaosystems,.*":
@@ -1490,6 +1492,8 @@ patternProperties:
     description: YSH & ATIL
   "^yones-toptech,.*":
     description: Yones Toptech Co., Ltd.
+  "^youritech,.*":
+    description: Shenzhen Youritech Technology Co., Limited
   "^ys,.*":
     description: Shenzhen Yashi Changhua Intelligent Technology Co., Ltd.
   "^ysoft,.*":
diff --git a/MAINTAINERS b/MAINTAINERS
index 48504f56a701..816cd293eb3d 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -4502,6 +4502,8 @@ BYTESATWORK MODULES AND BOARDS
 M:	Daniel Ammann <daniel.ammann@bytesatwork.ch>
 S:	Maintained
 F:	.github/workflows/checkpatch.yml
+F:	Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
+F:	Documentation/devicetree/bindings/display/panel/youritech,ili9806e.yaml
 
 C-MEDIA CMI8788 DRIVER
 M:	Clemens Ladisch <clemens@ladisch.de>
